python/pyspark/ml/feature.py

```diff
@@ -1030,6 +1030,87 @@ class Word2VecModel(JavaModel):
     """
 
 
+@inherit_doc
+class MinMaxScaler(JavaEstimator, HasInputCol, HasOutputCol):
+    """
+    Rescale each feature individually to a common range [min, max] linearly using column summary
+    statistics, which is also known as min-max normalization or Rescaling. The rescaled value for
+    feature E is calculated as,
+
+    Rescaled(e_i) = \frac{e_i - E_{min}}{E_{max} - E_{min}} * (max - min) + min
+
+    For the case E_{max} == E_{min}, Rescaled(e_i) = 0.5 * (max + min)
```

Comment:
Please copy full Scala doc: "Note that since zero values will probably be transformed to non-zero values, output of the transformer will be DenseVector even for sparse input."